### PR TITLE
Fixes #20839 No more double yolk demons

### LINKED ISSUE #20839: Multiple Laughter Demons from a single Bottle of Tickles
In a round of Ragin' Mages I purchased a bottle of Tickles with my last wizard point. I used it immediately after arriving on the station. I tried using it with the hotkey to use an item in your hand but nothing happened, so I quickly clicked on the item in my hand directly. After a bit of lag, I got the summoning message twice in quick succession, then ran off to do wizard stuff. After the round it turns out that there were two laughter demons summoned, both by me ( They both had the objective to "hug and tickle" me, so one wasn't made by one of the other wizards).

[I had 7 points worth of spells, along with a staff of change for a total of 9, last point was the bottle.](http://imgur.com/a/d5S6k) 

[Two Laughter demons on the end game screen, both with the objective to kill me.](http://imgur.com/a/qkLWu)

### PR DESCRIPTION
**Pull request by Supermichael777**
:cl:
fix: The wizards federation have instituted a strict 1 demon limit on demon bottles.
/:cl:

Fun fact this never actually checked whether it was used up

### CODE REVIEW

File: code/game/gamemodes/antag_spawner.dm [hunk 1]
```diff
@@ -233,6 +233,9 @@
 		user << "<span class='notice'>You should probably wait until you reach the station.</span>"
 		return
 	if(demon_candidates.len > 0)
+		if(used)
+			user << "<span class='notice'>Your already smashing the bottle, calm down.</span>"
```
Comment (Cheridan) [inline code/game/gamemodes/antag_spawner.dm]: >your

**Comment by Cruix** [comment]
Why does it pollCandidates before even checking the zlevel? You should probably move it below both checks.

File: code/game/gamemodes/antag_spawner.dm (hunk 1 above)
Comment (ike709) [inline code/game/gamemodes/antag_spawner.dm]: http://www.youryoure.com/

File: code/game/gamemodes/antag_spawner.dm [hunk 2]
```diff
@@ -228,11 +228,14 @@
 
 
 /obj/item/weapon/antag_spawner/slaughter_demon/attack_self(mob/user)
-	var/list/demon_candidates = pollCandidates("Do you want to play as a [initial(demon_type.name)]?", null, null, ROLE_ALIEN, 50)
 	if(user.z != 1)
 		user << "<span class='notice'>You should probably wait until you reach the station.</span>"
 		return
+	var/list/demon_candidates = pollCandidates("Do you want to play as a [initial(demon_type.name)]?", null, null, ROLE_ALIEN, 50)
```
Comment (phil235) [inline code/game/gamemodes/antag_spawner.dm]: might be safer to also check z level after pollCandidates(), maybe the wizard was initially on the station but moved to lavaland or drifted in space during the proc.

**Pull request merged**